example/js/js_image_layer.bzl

```diff
@@ -0,0 +1,260 @@
+"contains container helper functions for js_binary"
+
+load("@rules_pkg//:providers.bzl", "PackageFilegroupInfo", "PackageFilesInfo", "PackageSymlinkInfo")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@aspect_bazel_lib//lib:paths.bzl", "to_manifest_path")
+load("@rules_python//python:defs.bzl", "py_binary")
+
+# BAZEL_BINDIR has to be set to '.' so that js_binary preserves the PWD when running inside container.
+# See https://github.com/aspect-build/rules_js/tree/dbb5af0d2a9a2bb50e4cf4a96dbc582b27567155#running-nodejs-programs
+# for why this is needed.
+_LAUNCHER_TMPL = """
+export BAZEL_BINDIR=.
+source {executable_path}
```

Comment:
Why does it use `source` instead of `exec`?

Reply:
1- avoid $0 type of bugs. RUNFILES helper relies on that heavily. 
2- any future implicit reliance of executable_path being the entrypoint.

Reply:
1- avoid $0 type of bugs. RUNFILES helper relies on that heavily. 
2- any future implicit reliance of executable_path being the entrypoint.

Reply:
1- avoid $0 type of bugs. RUNFILES helper relies on that heavily. 
2- any future implicit reliance of executable_path being the entrypoint.

Reply:
1- avoid $0 type of bugs. RUNFILES helper relies on that heavily. 
2- any future implicit reliance of executable_path being the entrypoint.

Reply:
tried with exec. it doesn't work with it.